openbb_platform/providers/bls/openbb_bls/utils/helpers.py

```diff
@@ -0,0 +1,424 @@
+    import json  # noqa
+    from warnings import warn
+    from openbb_core.provider.utils.errors import EmptyDataError
+    from openbb_core.provider.utils.helpers import amake_request
+
+    symbols = series_ids.split(",") if isinstance(series_ids, str) else series_ids
+    if len(symbols) > 50:
+        warn(
+            "Max 50 symbols per request. Truncating to 50 symbols."
+            "Break the request into multiple queries to get more data."
+        )
+        symbols = symbols[:50]
+
+    headers = {"Content-type": "application/json"}
+    url = "https://api.bls.gov/publicAPI/v2/timeseries/data/"
+    payload = json.dumps(
+        {
+            k: v
+            for k, v in {
+                "seriesid": symbols,
+                "startyear": start_year,
+                "endyear": end_year,
+                "catalog": catalog,
+                "calculations": calculations,
+                "annualaverage": annual_average,
+                "aspects": aspects,
+                "registrationkey": api_key,
+            }.items()
+            if v
+        }
+    )
+    res = await amake_request(url=url, method="POST", headers=headers, data=payload)
+    results = res.get("Results", {}).get("series", [])  # type: ignore
+    messages = res.get("message", [])  # type: ignore
+    metadata: Dict = {}
+    data: List = []
+    for result in results:
+        seriesID = result.get("seriesID")
+        if not seriesID:
+            continue
+        catalog = result.get("catalog")
+        if catalog:
+            metadata.update({seriesID: catalog})
+        _data = result.get("data", [])
+        for _d in _data:
+            new_d: Dict = {}
+            year = _d.get("year", "")
+            month = _d.get("period", "").replace("M", "")
+            if month.startswith("A") or month in ("S01", "Q01"):
+                _date = year + "-01-01"
+            elif month == "S02":
+                _date = year + "-07-01"
+            elif month in ("S03", "Q05"):
+                _date = year + "-12-31"
+                month = "13"
+            elif month == "Q02":
+                _date = year + "-04-01"
+            elif month == "Q03":
+                _date = year + "-07-01"
+            elif month == "Q04":
+                _date = year + "-10-01"
+            else:
+                _date = year + "-12-31" if month == "13" else year + "-" + month + "-01"
+            new_d["symbol"] = seriesID
+            title = metadata[seriesID].get("series_title") if catalog else None
+            title = title + (" (Annual Average)" if month == "13" else "")
+            if title:
+                new_d["title"] = title
+            new_d["date"] = _date
+            value = _d.get("value")
+            if value and value != "-":
+                new_d["value"] = float(value)
+            else:
+                new_d["value"] = None
+            _latest = _d.get("latest") == "true"
+            new_d["latest"] = _latest
```

Comment:
Also things like here, you can just assign right to the dict instead of intermediate value that gets used in the next line

Reply:
Cleaned up.